crates/flox/src/commands/environment.rs

```diff
@@ -295,20 +295,31 @@ impl Activate {
             .context("couldn't restore cursor position")?;
         stderr.flush().context("could't flush stderr")?;
 
+        let activation_script = activation_path.join("activate").into_os_string();
+
+        // We don't have access to the current PS1 (it's not exported), so we
+        // can't modify it. Instead set FLOX_PROMPT_ENVIRONMENTS and let the
+        // activation script set PS1 based on that.
         let flox_prompt_environments = env::var("FLOX_PROMPT_ENVIRONMENTS")
             .map_or(prompt_name.clone(), |prompt_environments| {
                 format!("{prompt_environments} {prompt_name}")
             });
 
-        // We don't have access to the current PS1 (it's not exported), so we
-        // can't modify it. Instead set FLOX_PROMPT_ENVIRONMENTS and let the
-        // activation script set PS1 based on that.
-        let activation_script = activation_path.join("activate");
-        debug!("running activation script: {}", activation_script.display());
-        let error = Command::new(activation_script)
+        // TODO more sophisticated detection?
+        let shell = env::var("SHELL").unwrap();
+        let mut command = Command::new(&shell);
+        command
             .env("FLOX_PROMPT_ENVIRONMENTS", flox_prompt_environments)
-            .env("FLOX_ENV", activation_path)
-            .exec();
+            .env("FLOX_ENV", activation_path);
+
+        if shell.ends_with("bash") {
```

Comment:
non-blocking question: Does this only support `bash` for now?

Reply:
Yes see https://github.com/flox/flox/issues/396 for zsh